docs/proposals/24-shoot-oidc-issuer.md

```diff
@@ -0,0 +1,130 @@
+
+```yaml
+status:
+  advertisedAddresses:
+    ...
+    - name: service-account-issuer
+      url: https://<central.domain.name>/projects/<project-name>/shoots/<shoot-uid>/issuer
+```
+
+During the deletion flow of a shoot the previously synced configmap will be deleted from the Garden cluster.
+
+The contents of the described configmap will contain both the `/.well-known/openid-configuration` and the `/jwks` metadata. It is important to note that the issuer URL uses the shoot UID in its path. This is done so that a recreated shoot has a different issuer compared to its predecessor with the same name. See the following example.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: myproj--f924a208-1034-4aeb-84d9-b0184894a0cf
+  namespace: gardener-shoot-issuer
+  labels:
+    gardener.cloud/project: myproj
+    gardener.cloud/shootName: myshoot
+data:
+  openid-config: |
+    {
+        "issuer": "https://<central.domain.name>/projects/myproj/shoots/f924a208-1034-4aeb-84d9-b0184894a0cf/issuer",
+        "jwks_uri": "https://<central.domain.name>/projects/myproj/shoots/f924a208-1034-4aeb-84d9-b0184894a0cf/issuer/jwks",
+        "response_types_supported": [
+            "id_token"
+        ],
+        "subject_types_supported": [
+            "public"
+        ],
+        "id_token_signing_alg_values_supported": [
+            "RS256"
+        ]
+    }
+  jwks: |
+    {
+        "keys": [
+            {
+                "use": "sig",
+                "kty": "RSA",
+                "kid": "<THE_KEY_INDENTIFIER>",
+                "alg": "RS256",
+                "n": "<THE_PUBLIC_KEY>",
+                "e": "AQAB"
+            }
+        ]
+    }
+```
+
+### Metadata Server
+
+A new component called `gardener-metadata-server` will be introduced. This component will be maintained in a separate repository in order to decouple its development and release schedule from that of Gardener. It is natural that this component is deployed in the Garden cluster alongside with the the Gardener controlplane components. It may be managed by the `gardener-operator` or installed separately. The server will have minimal permissions and restricted access to the Garden cluster, i.e. it will only require read access for configmaps in the `gardener-shoot-issuer` namespace. The server will be publicly accessible and will serve the metadata information for different shoot clusters, i.e. when a `GET` request hits `https://<central.domain.name>/projects/myproj/shoots/f924a208-1034-4aeb-84d9-b0184894a0cf/issuer/.well-known/openid-configuration` the server should return the contents of `.data.openid-config` from the corresponding configmap. Since this server will be part of authentication flows it needs to be highly available and implemented with security and observability in mind.
```

Comment:
> It may be managed by the `gardener-operator` or installed separately.

I would make this more clear: "It will be managed by `gardener-operator`, or can be installed manually when not making use of it for managing the Gardener control plane."